[CI] DivTests testEvaluate {TestCase=<integer>, <dense_vector>} failing

**Build Scans:**
- [elasticsearch-periodic #12383 / openjdk26_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/ciyzxgchngzvo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.DivTests.testEvaluate {TestCase=<integer>, <dense_vector>}" -Dtests.seed=C38A510C0BEF0325 -Dtests.locale=en-LV -Dtests.timezone=Africa/Ndjamena -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.DivTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEvaluate%20%7BTestCase%3D%3Cinteger%3E%2C%20%3Cdense_vector%3E%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <[-4.785863E9, -2.690849E9, -4.8956846E9, -6.1919345E9, -2.371961E9, -6.5602284E9, -1.7327645E10, 8.36899E9, -1.9195963E9, 2.9070426E9, -Infinity, -4.934005E9, 2.815333E9, -2.1063508E10, 4.887776E9, 5.9701827E9, 2.6533332E9, -9.230275E9, 3.1988265E9, -1.6734151E10, -2.1244767E9, -2.7190564E9, 2.2449162E9, 2.0411253E10, -1.2641012E10, -3.2494277E9, 1.5960732E10, 2.1338479E9, 1.1807395E10, 2.446516E9, 2.245269E9, 9.479735E9, 2.9500498E9, -1.8117165E10, 1.3096477E10, -3.684471E9, 2.9669558E10, 2.3526444E9, 2.522802E9, -2.5717501E9, 1.5103275E10, 2.1441761E10, 1.9105775E9, 2.264882E9, -1.1578024E11, 2.5028086E9, -1.1745556E10, 2.3037683E9, -1.1089109E10, -1.1999906E10, -6.7300854E9, -3.5423565E9, -4.1961673E10, -4.7026427E9, 6.0273705E9, -3.2600133E9, 9.555726E10, -3.7286892E9, 2.2275382E9, -1.054811E11, 1.9080055E10, 2.0330412E10, -2.6051423E9, -7.687147E9, 2.6079114E9, 3.3379308E9, -1.6589583E10, 2.44188E9, 7.6281864E10, 2.7390753E9, -3.8788234E9, 9.7
[truncated]
```

**Issue Reasons:**
- [main] 1 consecutive failures in step openjdk26_checkpart3_java-matrix

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -506,15 +506,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testUnhollowOnUpdates
   issue: https://github.com/elastic/elasticsearch/issues/151100
-- class: org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.DivTests
-  method: testEvaluate {TestCase=<integer>, <dense_vector>}
-  issue: https://github.com/elastic/elasticsearch/issues/151114
-- class: org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.DivTests
-  method: testEvaluate {TestCase=<double>, <dense_vector>}
-  issue: https://github.com/elastic/elasticsearch/issues/151115
-- class: org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.DivTests
-  method: testEvaluate {TestCase=<long>, <dense_vector>}
-  issue: https://github.com/elastic/elasticsearch/issues/151116
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=index/100_field_name_length_limit/Test field name length limit synthetic source}
   issue: https://github.com/elastic/elasticsearch/issues/151117

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DivTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DivTests.java
@@ -218,25 +218,19 @@ public class DivTests extends AbstractScalarFunctionTestCase {
                 .withWarning("Line 1:1: java.lang.ArithmeticException: / by zero");
         }));
 
-        suppliers.addAll(
-            denseVectorScalarCases(
-                "Div",
-                (v, s) -> {
-                    Float sf = (Float) DataTypeConverter.convert(s, FLOAT);
-                    List<Float> result = v.stream().map(f -> f / sf).toList();
-                    // If the scalar denominator is zero or any element overflows to Infinity/NaN,
-                    // the evaluator throws ArithmeticException and returns null for the whole vector
-                    return result.stream().anyMatch(f -> Float.isInfinite(f) || Float.isNaN(f)) ? null : result;
-                },
-                (s, v) -> {
-                    Float sf = (Float) DataTypeConverter.convert(s, FLOAT);
-                    List<Float> result = v.stream().map(f -> sf / f).toList();
-                    // Large scalars (e.g. randomInt/randomLong) divided by near-zero vector elements
-                    // can overflow to Infinity; the evaluator throws and returns null for the whole vector
-                    return result.stream().anyMatch(f -> Float.isInfinite(f) || Float.isNaN(f)) ? null : result;
-                }
-            )
-        );
+        suppliers.addAll(denseVectorScalarCases("Div", (v, s) -> {
+            Float sf = (Float) DataTypeConverter.convert(s, FLOAT);
+            List<Float> result = v.stream().map(f -> f / sf).toList();
+            // If the scalar denominator is zero or any element overflows to Infinity/NaN,
+            // the evaluator throws ArithmeticException and returns null for the whole vector
+            return result.stream().anyMatch(f -> Float.isInfinite(f) || Float.isNaN(f)) ? null : result;
+        }, (s, v) -> {
+            Float sf = (Float) DataTypeConverter.convert(s, FLOAT);
+            List<Float> result = v.stream().map(f -> sf / f).toList();
+            // Large scalars (e.g. randomInt/randomLong) divided by near-zero vector elements
+            // can overflow to Infinity; the evaluator throws and returns null for the whole vector
+            return result.stream().anyMatch(f -> Float.isInfinite(f) || Float.isNaN(f)) ? null : result;
+        }));
 
         // Overflows
         suppliers.addAll(

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DivTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DivTests.java
@@ -221,8 +221,20 @@ public class DivTests extends AbstractScalarFunctionTestCase {
         suppliers.addAll(
             denseVectorScalarCases(
                 "Div",
-                (v, s) -> v.stream().map(f -> f / (Float) DataTypeConverter.convert(s, FLOAT)).toList(),
-                (s, v) -> v.stream().map(f -> ((Float) DataTypeConverter.convert(s, FLOAT) / f)).toList()
+                (v, s) -> {
+                    Float sf = (Float) DataTypeConverter.convert(s, FLOAT);
+                    List<Float> result = v.stream().map(f -> f / sf).toList();
+                    // If the scalar denominator is zero or any element overflows to Infinity/NaN,
+                    // the evaluator throws ArithmeticException and returns null for the whole vector
+                    return result.stream().anyMatch(f -> Float.isInfinite(f) || Float.isNaN(f)) ? null : result;
+                },
+                (s, v) -> {
+                    Float sf = (Float) DataTypeConverter.convert(s, FLOAT);
+                    List<Float> result = v.stream().map(f -> sf / f).toList();
+                    // Large scalars (e.g. randomInt/randomLong) divided by near-zero vector elements
+                    // can overflow to Infinity; the evaluator throws and returns null for the whole vector
+                    return result.stream().anyMatch(f -> Float.isInfinite(f) || Float.isNaN(f)) ? null : result;
+                }
             )
         );
 

